requirements.txt

```diff
@@ -1,225 +1,32 @@
-aiohttp==3.9.5
-aiosignal==1.3.1
-altair==5.3.0
-annotated-types==0.7.0
-anyio==4.4.0
-appdirs==1.4.4
-appnope==0.1.4
-argon2-cffi==23.1.0
-argon2-cffi-bindings==21.2.0
-arrow==1.3.0
-asttokens==2.4.1
-async-lru==2.0.4
-atlassian-python-api==3.41.14
-attrs==23.2.0
-Authlib==1.3.1
-Automat==22.10.0
-Babel==2.15.0
-backoff==2.2.1
-beautifulsoup4==4.12.3
-bleach==6.1.0
-blinker==1.8.2
-cachetools==5.3.3
-certifi==2024.6.2
-cffi==1.16.0
-chardet==5.2.0
-charset-normalizer==3.3.2
-click==8.1.7
-comm==0.2.2
-constantly==23.10.4
-cryptography==42.0.8
-cssselect==1.2.0
-dataclasses-json==0.6.7
-datasets==2.20.0
-debugpy==1.8.1
-decorator==5.1.1
-deepdiff==7.0.1
-defusedxml==0.7.1
-Deprecated==1.2.14
-dill==0.3.8
-dirtyjson==1.0.8
-distro==1.9.0
-emoji==2.12.1
-executing==2.0.1
-fastjsonschema==2.20.0
-filelock==3.15.1
-filetype==1.2.0
-fqdn==1.5.1
-frozenlist==1.4.1
-fsspec==2024.5.0
-gitdb==4.0.11
-GitPython==3.1.43
-greenlet==3.0.3
-h11==0.14.0
-html2text==2024.2.26
-httpcore==1.0.5
-httpx==0.27.0
-huggingface-hub==0.23.4
-hyperlink==21.0.0
-idna==3.7
-incremental==22.10.0
-# ipykernel==6.29.4
-# ipython==8.18.0
-isoduration==20.11.0
-itemadapter==0.9.0
-itemloaders==1.3.1
-jedi==0.19.1
-Jinja2==3.1.4
-jmespath==1.0.1
-joblib==1.4.2
-json5==0.9.25
-jsonpatch==1.33
-jsonpath-python==1.0.6
-jsonpointer==3.0.0
-jsonschema==4.22.0
-jsonschema-specifications==2023.12.1
-jupyter-events==0.10.0
-jupyter-lsp==2.2.5
-jupyter_client==8.6.2
-jupyter_core==5.7.2
-jupyter_server==2.14.1
-jupyter_server_terminals==0.5.3
-jupyterlab==4.2.2
-jupyterlab_pygments==0.3.0
-jupyterlab_server==2.27.2
-langchain==0.2.5
-langchain-community==0.2.5
-langchain-core==0.2.9
-langchain-openai==0.1.9
-langchain-text-splitters==0.2.1
-langdetect==1.0.9
-langsmith==0.1.81
+# Note: Use python 3.9
```

Comment:
Great to note this. In the future we should update to a newer Python and migrate to Pixi or uv to more clearly define the Python dependency.